specs/recognition/image_recognition_spec.py

```diff
@@ -0,0 +1,21 @@
+from flask import current_app
+from mamba import description, before, it
+from expects import expect, be_none
+from doublex import Mock, ANY_ARG
+from specs.spec_helpers import Context
+from application.controllers.artifacts_controller import no_content
+from application.recognition.image_recognition import _work_asynchronously
+from specs.factories.image_recognition import vision_api_response
+from unittest.mock import patch
+
+with description('image recognition called during create') as self:
+    with before.each:
+        self.context = Context()
+        with Mock() as artifact_mock:
+            artifact_mock.update(ANY_ARG).returns(no_content())
+            artifact_mock.tags = []
+        self.artifact = artifact_mock
+    with it('Adds Tags to artifact'):
+        with patch('application.recognition.image_recognition._call_google_api',
+                   return_value=vision_api_response()):
+            expect(_work_asynchronously('files.askir.me/test.jpg', self.artifact)).to(be_none)

```

Comment:
Just a note here btw, in general testing private methods is considered to be bad style. You should only test the behaviour of the method not the implementation. But since you are only doing this to test the asynchronous nature, this is a case where testing a private method is necessary.

I'm not sure if you knew this so I just wanted to mention it.

Reply:
I tried out testing the auto_add_tags function, but it appears that mocks do not carry over onto the worker thread, or at the very least never hear back from the worker thread, so running the tests including the thread spawn does not help.